Improve contact timeline readability with intuitive date grouping

kuzmany opened the pull request:
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | ❌
| New feature/enhancement? (use the a.x branch)      | ✔️
| Deprecations?                          | ❌
| BC breaks? (use the c.x branch)        | ❌
| Automated tests included?              | ❌
| Related user documentation PR URL      | N/A
| Related developer documentation PR URL | N/A
| Issue(s) addressed                     | Improves timeline UX with better organization

## Description

This PR enhances the contact timeline display by grouping events by days with improved visual design:

**Key Features:**
- **Day Grouping**: Events are now grouped under "Today", "Yesterday", and formatted dates (e.g., "23. 05. 2025")
- **Enhanced Date Headers**: Styled with warm gray label badges and time icons for better visual hierarchy
- **Time-Only Display**: Individual events show only time since date is in the header
- **Template-Only Implementation**: Simple Twig logic without controller changes
- **Responsive Design**: Maintains all existing functionality (filtering, sorting, pagination)

**Visual Improvements:**
- Centered date headers with label styling
- Time icons for visual consistency
- Clean separation between date groups
- Better scanning and readability

<img width="584" height="530" alt="image" src="https://github.com/user-attachments/assets/86550090-b12b-4be1-92ec-db1a77c2f7a8" />


---
### 📋 Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally
2. Navigate to any contact's timeline (Contacts > [Contact] > Timeline)
3. Verify events are grouped by days:
   - Today's events appear under "Today" header
   - Yesterday's events under "Yesterday" header
   - Older events under formatted date headers (e.g., "23. 05. 2025")
4. Confirm time-only display in timestamp column
5. Test that filtering, sorting, and pagination still work correctly
6. Verify visual consistency with warm gray date header badges

kuzmany committed:
feat: group timeline events by days (Today, Yesterday, dates)

Timeline now displays events grouped by days with clear date headers:
- Today/Yesterday labels for recent events
- Formatted dates (j. m. Y) for older events
- Time-only display in timestamp column
- Simple template-only implementation using Twig date logic

kuzmany committed:
feat: enhance timeline date headers with label styling

- Replace basic date headers with warm gray label badges
- Add time icon (ri-time-line) for visual consistency
- Center align date headers for better appearance
- Use larger label size with normal font weight
- Maintain grouped display: Today, Yesterday, formatted dates

kuzmany committed:
refine: adjust timeline date header label styling

- Change label size from lg to md for better proportions
- Increase font size to 13px for improved readability
- Switch icon from calendar to time (ri-time-line) for consistency
- Maintain warm gray label styling with normal font weight

kuzmany commented:
Inspiration was datafa.st contact activity 
<img width="1335" height="707" alt="image" src="https://github.com/user-attachments/assets/b554302e-f07b-4754-aace-baa8626e90df" />

andersonjeccel reviewed (changes requested):
There are hardcoded words, please switch to translation strings

kuzmany committed:
Merge remote-tracking branch 'upstream/7.x' into private-pr-6508-group-timeline-events-by-days-enhanced-date

kuzmany committed:
Add dateToTextShort method for clean date display

- Add dateToTextShort Twig function to DateExtension
- Implement toTextShort method in DateHelper using DateTimeHelper::getTextDate()
- Replace hardcoded date logic in Timeline/_list.html.twig with dateToTextShort
- Returns "Today", "Yesterday" (translated) or formatted date for other dates
- Provides clean, reusable solution for date labels across the application

kuzmany committed:
Add comprehensive unit tests for dateToTextShort method

DateExtensionTest.php:
- Update function count from 8 to 9 to include dateToTextShort
- Add dateToTextShort to function name assertions
- Add testToTextShort() method for basic functionality testing

DateHelperTest.php:
- Update translator mock to handle today/yesterday translations with %time% parameter
- Add testToTextShortWithToday() - tests 'today' detection and translation
- Add testToTextShortWithOlderDate() - tests fallback to formatted date
- Add testToTextShortWithEmptyDateTime() - tests empty input edge case
- Use mocking strategy consistent with existing toText tests

kuzmany commented:
Fixed. 
Unit test fix later

kuzmany committed:
Merge branch '7.x' into private-pr-6508-group-timeline-events-by-days-enhanced-date

jos0405 reviewed (approved):
Does what is sais, really nice new UI. Thank you!

<img width="1896" height="1056" alt="image" src="https://github.com/user-attachments/assets/631dc392-be36-42da-a2a9-4fb3c08f0499" />

kuzmany committed:
Merge branch '7.x' into private-pr-6508-group-timeline-events-by-days-enhanced-date

kuzmany committed:
Fix DateHelperTest by mocking toLocalString method

The testToTextShortWithOlderDate test was failing because when
getTextDate() returns false, the code calls format() which uses
toLocalString() on the mocked DateTimeHelper. Without mocking
toLocalString(), it returned null causing a type error.

kuzmany committed:
Merge branch '7.x' into private-pr-6508-group-timeline-events-by-days-enhanced-date

shinde-rahul reviewed (commented):
Works as expected and passes user testing from my side.

There are still a few failing pipelines, though. Please check and fix those before this can move forward.

shinde-rahul reviewed (approved):
Approving this from the user testing side.

Bastian2718 reviewed (approved):
Works as expected
<img width="1678" height="534" alt="Image" src="https://github.com/user-attachments/assets/181c4f7f-0281-4992-b5cb-e37c00e51d21" />

escopecz committed:
Merge branch '7.x' into private-pr-6508-group-timeline-events-by-days-enhanced-date
